### DISCUSSION TITLE
Inheritance

### DISCUSSION THREAD

**Discussion by timothyjbrooks** [cwendt94/espn-api#371] [ESPN API]
I'm trying to write a TeamReport class that inherits the Team class and adds some attributes as well as defines some calculation methods. I see that the constructor for Team passes in "data" but it's a bit convoluted to me as to what that consists of. I'm fairly new to python and much more familiar with strongly typed languages so this is confusing me and taking longer to figure out than I'd like. If you have any pointers please let me know.

```
class TeamReport(Team):
  def __init__(self, team, rank, pointsLeftOnBench, projectedRecord, playoffChance, winningChance, overachieversCount, underachieversCount, margin ):
    self.rank = rank
    self.pointsLeftOnBench = pointsLeftOnBench
    self.projectedRecord = projectedRecord
    self.playoffChance = playoffChance
    self.winningChance = winningChance
    self.overachieversCount = overachieversCount
    self.underachieversCount = underachieversCount
    self.margin = margin
    super().__init__(team.data, team.roster, team.member, team.schedule, team.year)
```

**Comment by cwendt94** [comment]
Understandable, the data returned from ESPN is a lot and unfortunately not easy to define types/structures for the data. The best way to really understand the data being passed is looking at the raw response. Hopefully in the future I can go through and create types for a lot of this data.

To see all the data passed into the Teams class you can make this call with the League object `league.espn_request.get_league()`. This call gets all the initial data for the League class. [Here](https://github.com/cwendt94/espn-api/blob/master/espn_api/base_league.py#L45) is the code that splits out some of that data and initializes the team class
